### PR TITLE
TTV code cleanup

### PR DESCRIPTION
**Pull request by silicons**

### CODE REVIEW

File: code/game/objects/items/devices/transfer_valve.dm [hunk 1]
```diff
@@ -76,21 +76,21 @@
...
-	if ( usr.stat || usr.restrained() )
+	if(!usr.canUseTopic())
 		return
-	if (src.loc == usr)
+	if(loc == usr)
 		if(tank_one && href_list["tankone"])
 			split_gases()
 			valve_open = FALSE
-			tank_one.loc = get_turf(src)
+			tank_one.drop_location()
```
Comment (Cyberboss) [inline code/game/objects/items/devices/transfer_valve.dm]: tank_one.forceMove(drop_location())

File: code/game/objects/items/devices/transfer_valve.dm [hunk 2]
```diff
@@ -76,21 +76,21 @@
...
 			tank_one = null
 			update_icon()
 			if((!tank_two || tank_two.w_class < WEIGHT_CLASS_BULKY) && (w_class > WEIGHT_CLASS_NORMAL))
 				w_class = WEIGHT_CLASS_NORMAL
 		else if(tank_two && href_list["tanktwo"])
 			split_gases()
 			valve_open = FALSE
-			tank_two.loc = get_turf(src)
+			tank_two.drop_location()
```
Comment (Cyberboss) [inline code/game/objects/items/devices/transfer_valve.dm]: tank_two.forceMove(drop_location())

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/transfer_valve.dm [hunk 3]
```diff
@@ -99,24 +99,26 @@
 			toggle_valve()
 		else if(attached_device)
 			if(href_list["rem_device"])
-				attached_device.forceMove(get_turf(src))
+				attached_device.drop_location()
```
Comment (Cyberboss) [inline code/game/objects/items/devices/transfer_valve.dm]: forceMove

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/transfer_valve.dm [hunk 4]
```diff
@@ -76,21 +76,21 @@
 
 /obj/item/device/transfer_valve/Topic(href, href_list)
 	..()
-	if ( usr.stat || usr.restrained() )
+	if(!usr.canUseTopic())
```
Comment (optimumtact) [inline code/game/objects/items/devices/transfer_valve.dm]: do you need to set be_close to true or is it on by default I don't recall I am a brainlet

Comment (optimumtact) [reply code/game/objects/items/devices/transfer_valve.dm]: HELLO

Comment (kevinz000) [reply code/game/objects/items/devices/transfer_valve.dm]: SORRY

Comment (kevinz000) [reply code/game/objects/items/devices/transfer_valve.dm]: `
/mob/living/carbon/human/canUseTopic(atom/movable/M, be_close = 0)
	if(incapacitated() || lying )
		return
	if(!Adjacent(M) && (M.loc != src))
		if((be_close == 0) && (dna.check_mutation(TK)))
			if(tkMaxRangeCheck(src, M))
				return 1
		return
	return 1
`

**Pull request merged**